libcodechecker/server/api/report_server.py

```diff
@@ -1758,51 +1758,108 @@ def removeRunResults(self, run_ids):
-                    # Commmit the can_delete flag.
-                    session.commit()
+            if run_locks:
+                raise shared.ttypes.RequestFailed(
+                    shared.ttypes.ErrorCode.DATABASE,
+                    "Can not remove results because the following runs "
+                    "are locked: {0}".format(
+                        ', '.join([r[0] for r in run_locks])))
 
-                    session.query(Run)\
-                        .filter(Run.id == run_id)\
-                        .delete(synchronize_session=False)
+            try:
+                diff_hashes = None
+                if cmp_data:
+                    diff_hashes, run_ids = self._cmp_helper(session,
+                                                            run_ids,
+                                                            report_filter,
+                                                            cmp_data)
+                    if not diff_hashes:
+                        # There is no difference.
+                        return True
+
+                filter_expression = process_report_filter(session,
+                                                          report_filter)
+
+                q = session.query(Report.id) \
+                    .outerjoin(File, Report.file_id == File.id) \
+                    .outerjoin(ReviewStatus,
+                               ReviewStatus.bug_hash == Report.bug_id) \
+                    .filter(filter_expression)
 
-                    # Delete files and contents that are not present
-                    # in any bug paths.
-                    db_cleanup.remove_unused_files(session)
-                    session.commit()
-                    session.close()
+                if run_ids:
+                    q = q.filter(Report.run_id.in_(run_ids))
+
+                if cmp_data:
+                    q = q.filter(Report.bug_id.in_(diff_hashes))
+
+                reports_to_delete = [r[0] for r in q]
+                if len(reports_to_delete) != 0:
+                    session.query(Report) \
+                        .filter(Report.id.in_(reports_to_delete)) \
+                        .delete(synchronize_session=False)
 
+                # Delete files and contents that are not present
+                # in any bug paths.
+                db_cleanup.remove_unused_files(session)
+                session.commit()
+                session.close()
+                return True
             except Exception as ex:
-                LOG.error("Failed to remove run: " + str(run_id))
+                session.rollback()
+                LOG.error("Database cleanup failed.")
                 LOG.error(ex)
-                failed = True
+                return False
 
-        return not failed
+    @exc_to_thrift_reqfail
+    @timeit
+    def removeRun(self, run_id):
+        self.__require_store()
+
+        # Remove all reports from the run.
+        ret = self.removeRunReports([run_id], None, None)
```

Comment:
Do we need to remove the run reports separately before removing the run?
The cascade delete should remove the reports for a run.